test/convolve.cpp

```diff
@@ -804,3 +805,335 @@ TEST(DISABLED_ConvolveLargeDim3D, CPP) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt7711_s77_p11_d11.test"),
+        dim4(7, 7), dim4(1, 1), dim4(1, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Strided_sig1010_filt33_s11_p11_d22) {
+    convolve2stridedTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3311_s11_p11_d22.test"),
+        dim4(1, 1), dim4(1, 1), dim4(2, 2));
+}
+
+TYPED_TEST(ConvolveStrided, Gradient_sig1010_filt33_s11_p11_d22) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3311_s11_p11_d22.test"),
+        dim4(1, 1), dim4(1, 1), dim4(2, 2));
+}
+
+TYPED_TEST(ConvolveStrided, Strided_sig1010_filt33_s11_p11_d33) {
+    convolve2stridedTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3311_s11_p11_d33.test"),
+        dim4(1, 1), dim4(1, 1), dim4(3, 3));
+}
+
+TYPED_TEST(ConvolveStrided, Gradient_sig1010_filt33_s11_p11_d33) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3311_s11_p11_d33.test"),
+        dim4(1, 1), dim4(1, 1), dim4(3, 3));
+}
+
+TYPED_TEST(ConvolveStrided, Strided_sig1010_filt35_s11_p11_d11) {
+    convolve2stridedTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3511_s11_p11_d11.test"),
+        dim4(1, 1), dim4(1, 1), dim4(1, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Gradient_sig1010_filt35_s11_p11_d11) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3511_s11_p11_d11.test"),
+        dim4(1, 1), dim4(1, 1), dim4(1, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Strided_sig1010_filt53_s11_p11_d11) {
+    convolve2stridedTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt5311_s11_p11_d11.test"),
+        dim4(1, 1), dim4(1, 1), dim4(1, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Gradient_sig1010_filt53_s11_p11_d11) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt5311_s11_p11_d11.test"),
+        dim4(1, 1), dim4(1, 1), dim4(1, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Strided_sig1010_filt35_s31_p11_d21) {
+    convolve2stridedTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3511_s31_p11_d21.test"),
+        dim4(3, 1), dim4(1, 1), dim4(2, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Gradient_sig1010_filt35_s31_p11_d21) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig101011_filt3511_s31_p11_d21.test"),
+        dim4(3, 1), dim4(1, 1), dim4(2, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Strided_sig81032_filt3334_s11_p11_d11) {
+    convolve2stridedTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig81032_filt3334_s11_p11_d11.test"),
+        dim4(1, 1), dim4(1, 1), dim4(1, 1));
+}
+
+TYPED_TEST(ConvolveStrided, Gradient_sig81032_filt3334_s11_p11_d11) {
+    convolve2GradientTest<TypeParam>(
+        string(TEST_DIR "/convolve/sig81032_filt3334_s11_p11_d11.test"),
+        dim4(1, 1), dim4(1, 1), dim4(1, 1));
+}
```

Comment:
Is batch support available in all new convolution functions ? If yes, we should add tests for that also.

Reply:
the numbers are confusing me a bit, can you please tell if there are batch tests among these.